lib/compiler-singlepass/src/machine.rs

```diff
@@ -400,12 +400,11 @@ impl Machine {
         // the end address of the 0th local, not at the start address, so we add `8` bytes to fix
         // this up.
         self.locals_offset = MachineStackOffset(static_area_size + 8);
-        let locals_size = (n as usize).saturating_sub(Self::LOCAL_REGISTERS.len()) * 8;
 
         // Allocate the stack, without actually writing to it.
         a.emit_sub(
             Size::S64,
-            Location::Imm32((static_area_size + locals_size) as _),
+            Location::Imm32(static_area_size as u32 + n_params),
```

Comment:
Worthwhile to cross-corelate/debug_assert that finite-wasm’s view of `static_area_size` and `n_params` is in any way similar to `wasmer`s.

Reply:
Hmmm so for now wasmer always allocates in qwords, whereas its current usage of finite-wasm accounts in actual number of bytes, so the numbers would never match.

I think it makes sense for it to set finite-wasm in number of bytes, because this is what will be exposed by the protocol. Ideally it’d use less memory by allocating less than a full qword per local, but I don’t think it’s actually a pressing thing to do.

Also, with the registers spilled being dependent on what exactly is being run, I don’t think even an "between x and 2x" assert would work. Maybe a "between x-32 and 2x+32" sanity-check would actually pass, but then I think the assert would be granting us undeserved peace of mind, as it wouldn’t really check much.